session.go

```diff
@@ -226,7 +235,12 @@ runLoop:
 			// This is a bit unclean, but works properly, since the packet always
 			// begins with the public header and we never copy it.
 			putPacketBuffer(p.publicHeader.Raw)
-		case l := <-s.aeadChanged:
+		case l, ok := <-aeadChanged:
+			if !ok {
+				s.handshakeComplete = true
+				aeadChanged = nil // prevent this case from ever being selected again
+				continue
```

Comment:
I don't have the full context right now, but are you sure about this continue? I think that completing the handshake unblock some things, or at the least change some timeouts, so that we should go through the full loop once?